core/dbt/context/providers.py

```diff
@@ -606,6 +606,29 @@ def resolve(self, source_name: str, table_name: str):
         return self.Relation.create_from_source(target_source)
 
 
+class RuntimeUnitTestSourceResolver(RuntimeSourceResolver):
```

Comment:
I think we're overriding the only method `RuntimeSourceResolver` provides & not using super(), so it may make more sense to subclass `BaseSourceResolver` here.

Reply:
Done.